neo/Network/P2P/LocalNode.cs

```diff
@@ -174,6 +223,50 @@ private void OnSendDirectly(IInventory inventory)
             Connections.Tell(inventory);
         }
 
+        protected override void TcpDisconnect(DisconnectReason reason)
+        {
+            var disconnectMessage = CreateDisconnectMessage(reason);
+            var command = Tcp.Write.Create(ByteString.FromBytes(disconnectMessage.ToArray()));
+
+            Sender.Tell(new Tcp.Register(ActorRefs.NoSender));
+            Sender.Ask(command).ContinueWith(t => Sender.Tell(Tcp.Abort.Instance));
+        }
+
+        protected override void WsDisconnect(WebSocket ws, DisconnectReason reason)
+        {
+            var disconnectMessage = CreateDisconnectMessage(reason);
+            ArraySegment<byte> segment = new ArraySegment<byte>(disconnectMessage.ToArray());
+
+            ws.SendAsync(segment, WebSocketMessageType.Binary, true, CancellationToken.None).PipeTo(Self,
+                failure: ex => new Tcp.ErrorClosed(ex.Message));
+            ws.CloseAsync(WebSocketCloseStatus.NormalClosure, "close ws", CancellationToken.None);
+        }
+
+        private Message CreateDisconnectMessage(DisconnectReason reason)
+        {
+            NetworkAddressWithTime[] networkAddresses = GetRandomConnectedPeers(AddrPayload.MaxCountToSend);
+            var payload = DisconnectPayload.Create(reason, networkAddresses.ToByteArray());
+            return Message.Create(MessageCommand.Disconnect, payload);
+        }
```

Comment:
Do you want to send a list of nodes for all the reasons? I think it may only be necessary to send when the reason is `MaxConnectionReached` or `MaxConnectionPerAddressReached`.

Reply:
It's true, but currently, `CreateDisconnectMessage` only invoked by those two cases. I think when there are different calling scenarios in the future, we just make a distinction here.